src/rt/sched/cown.h

```diff
@@ -967,16 +963,103 @@ namespace verona::rt
       return false;
     }
 
+    inline bool check_unmute_message(Alloc* alloc, MessageBody* msg)
+    {
+      if (msg->behaviour != nullptr)
+        return false;
+
+      for (size_t i = 0; i < msg->count; i++)
+      {
+        auto* cown = msg->cowns[i];
+        if (cown == nullptr)
+          break;
+
+        Systematic::cout() << "Unmute cown " << cown << std::endl;
+        cown->backpressure_transition(Priority::Normal);
+        Cown::release(alloc, cown);
+      }
+
+      alloc->dealloc(msg->cowns, msg->count * sizeof(Cown*));
+      alloc->dealloc<sizeof(MessageBody)>(msg);
+
+      return true;
+    }
+
     /// Mute the senders participating in this message if a backpressure scan
     /// set the mutor during the behaviour. If false is returned, the caller
     /// must reschedule the senders and deallocate the senders array.
-    inline bool apply_backpressure(Cown** senders, size_t count)
+    inline bool apply_backpressure(
+      Alloc* alloc, EpochMark epoch, Cown** senders, size_t count)
     {
-      if (Scheduler::local()->mutor == nullptr)
+      auto* mutor = Scheduler::local()->mutor;
+      if (mutor == nullptr)
         return false;
 
-      Scheduler::local()->mute(senders, count);
       Scheduler::local()->mutor = nullptr;
+      size_t muting_count = 0;
+      for (size_t i = 0; i < count; i++)
+      {
+        auto* cown = senders[i];
+        auto bp = cown->bp_state.load(std::memory_order_relaxed);
+        const bool high_priority = bp.high_priority();
+        yield();
+        assert(bp.priority() != Priority::Low);
+
+        // The cown may only be muted if its priority is normal and its epoch
+        // mark is not `SCANNED`. Muting a scanned cown may result in the leak
+        // detector collecting the cown while it is muted.
+        if (high_priority || (cown->get_epoch_mark() == EpochMark::SCANNED))
+        {
+          cown->schedule();
```

Comment:
what if the list has multiple high priority? do you schedule all of them at the same time?

Reply:
Any of the cowns with a high priority will be scheduled, yes.